qt/python/mantidqt/mantidqt/widgets/fitpropertybrowser/fitpropertybrowser.py

```diff
@@ -117,9 +117,16 @@ def _get_allowed_spectra(self):
                     # we don't want to include the fit workspace in our selection
                     if ws_name in output_wsnames:
                         continue
-                    spectrum_list = [artist.spec_num for artist in artists]
-                    spectrum_list = sorted(list(set(spectrum_list)))
-                    allowed_spectra[ws_name] = spectrum_list
+                    #loop through arists and get relevant spec numbers if spec_num represents a spectrum as opposed to a bin axes.
+                    spectrum_list = []
```

Comment:
Nice catch of fixing this bug.

Could we still use a list comprehension here? They are generally the fastest way to build lists and this is still as readable I think:
```python
spectrum_list = [artist.spec_num for artist in artists if artist.is_spec]
```

Reply:
Happy to add this, agree it is more concise and just as readable